warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/shard/ShardedDataTypeHandler.java

```diff
@@ -1141,21 +1150,38 @@ protected void createShardFieldIndexColumn(RawRecordContainer event, Multimap<Bu
      *            flag for the delete count
      * @return a value
      */
-    private Value createUidArray(String uid, boolean isDeleted) {
-        
-        // Create a UID object for the Value
-        Builder uidBuilder = Uid.List.newBuilder();
-        uidBuilder.setIGNORE(false);
-        
-        if (isDeleted)
+    protected Value createUidArray(String uid, boolean isDeleted) {
+        return new Value(createUidList(uid, isDeleted).toByteArray());
+    }
+
+    /**
+     * Create a {@link Uid.List} given a uid and delete mode flag
+     *
+     * @param uid        an event uid
+     * @param deleteMode a flag
+     * @return a Uid.List
+     */
+    protected Uid.List createUidList(String uid, boolean deleteMode) {
+        Uid.List.Builder uidBuilder = Uid.List.newBuilder();
+
+        //  delete mode takes precedent over disabled uid creation
+        if (deleteMode) {
+            uidBuilder.setIGNORE(false);
             uidBuilder.setCOUNT(-1);
-        else
+            uidBuilder.addUID(uid);
+        } else if (!getShardIndexCreateUids()) {
+            //  uid not created
+            uidBuilder.setIGNORE(true);
             uidBuilder.setCOUNT(1);
-        
-        uidBuilder.addUID(uid);
-        
-        Uid.List uidList = uidBuilder.build();
-        return new Value(uidList.toByteArray());
+            return uidBuilder.build();
```

Comment:
Oh missed this before I hit approve. This return can be removed.

Reply:
Missed that, fixed in next commit